IsogeomGenerator/generate_isogeom.py

```diff
@@ -0,0 +1,332 @@
+
+    # set visit only mode options
+    visit_parser = subparsers.add_parser('visit', help='Only generate the ' +
+                                         'isosurface mesh files using VisIt.')
+    set_visit_only_options(visit_parser)
+    set_shared_options(visit_parser)
+    visit_parser.set_defaults(which='visit')
+
+    # set moab only mode options
+    moab_parser = subparsers.add_parser('moab', help='Only generate the ' +
+                                        'DAGMC-compliant geometry with MOAB ' +
+                                        'starting from the VisIt mesh files.')
+    set_shared_options(moab_parser, moab=True)
+    set_moab_only_options(moab_parser)
+    moab_parser.set_defaults(which='moab')
+
+    args = parser.parse_args()
+    return args
+
+
+def check_level_gen(args):
+    """Check that correct args were supplied if generating level
+    """
+    if (args.extN is None) or (args.N is None):
+        raise RuntimeError("Min/Max level values (-lx) and number of levels " +
+                           "(-N) must be set to use --generatelevels option.")
+
+
+def get_levels(args, g):
+    """Get level information depending on supplied args.
+
+    Input:
+    ------
+        args: argparse ArgumentParser
+        g: IsoVolume instance
+    """
+    # collect level information:
+    if args.levelfile is not None:
+        # option 1: read from file
+        g.read_levels(args.levelfile[0])
+    elif args.levelvalues is not None:
+        # option 2: set values at run time
+        g.assign_levels(args.levelvalues)
+    elif args.generatelevels is not None:
+        # option 3: generate levels
+        check_level_gen(args)
+        minN = min(args.extN)
+        maxN = max(args.extN)
+        N = args.N[0]
+        g.generate_levels(N, minN, maxN, mode=args.generatelevels[0])
+    else:
+        raise RuntimeError("Mode for setting level information is not " +
+                           "recognized")
+
+
+def process_tags(tags):
+    """Convert the provided tag information to a dictionary to pass to the
+    geometry creation stepself.
+
+    Input:
+    -----
+        tags: list of lists, [[tagname1, tagval1], [tagname2, tagval2], ...]
+
+    Return:
+    -------
+        tagdict: dict, key=TAGNAME, value=TAGVALUE
+    """
+    tagdict = {}
+    for tagset in tags:
+        key = tagset[0]
+        val = float(tagset[1])
+        tagdict[key] = val
+
+    return tagdict
+
+
+def main():
+
+    # get args
+    args = parse_arguments()
+    print(args)
+
+    # create instance
+    g = v.IsoVolume()
+
+    # get level info regardless of mode
+    get_levels(args, g)
+
+    # get database information
+    db = os.getcwd() + '/' + args.db[0]
+
+    # run steps depending on mode
+    mode = args.which
+    if mode == 'full' or mode == 'visit':
+        # generate isosurfaces
+        meshfile = args.meshfile[0]
+        dataname = args.dataname[0]
+        g.generate_volumes(meshfile, dataname, dbname=db)
```

Comment:
why provide only one variable name ?
```suggestion
        g.generate_volumes(meshfile, dataname, db)
```

Reply:
maybe directly use `args.meshfile[0]` and `args.dataname[0]`